types/marked/index.d.ts

```diff
@@ -119,7 +119,7 @@ declare namespace marked {
 
     type TokensList = Token[] & {
         links: {
-            [key: string]: { href: string; title: string; }
+            [key: string]: { href: string; title: string; } | undefined
```

Comment:
When I look at the sources, I can't see any place where a `link` key is set without a value. This is the location I'm talking about: https://github.com/markedjs/marked/blob/master/lib/marked.js#L441

Is there any other location I'm missing?

Reply:
I understand what you mean but it is kind of alien to me to add `undefined` type to index signatures.

If you have an array, for example, you know that the indexes you access may be `undefined` at any time, I'm not sure that I've already seen an instance where someone added `undefined` to an array type to have an explicit check for non-existing indexes. Same goes here. It's just convenience of usage, I suppose.

Reply:
I agree with @BendingBender here as well. While I'm sympathetic to the issue here, I don't think that adding this is the right way to fix it given that it invalidates any object's type definition.

Reply:
@BendingBender @worr 
Thank you for your explanation. I am a Typescript beginner and I think I misunderstood this part. In the array example below, I have checked there is no error, and I have learned that it is a normal operation.

```typescript
const user: Array<{ name: string }> = []
const userName = user[4].name
```